File: fotostorio-microservices/Services/Ordering/Ordering.API/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Ordering.API.Models;

public class Order : BaseEntity
{
    public Order()
    {
    }

    public Order(List<OrderItem> items, string buyerEmail, Address sendToAddress, decimal total, string paymentIntentId)
    {
        Items = items;
        BuyerEmail = buyerEmail;
        SendToAddress = sendToAddress;
        Total = total;
        PaymentIntentId = paymentIntentId;
    }

    public string BuyerEmail { get; set; }

    public DateTimeOffset OrderDate { get; set; } = DateTimeOffset.Now;

    [Required]
    public Address SendToAddress { get; set; }

    [Required]
    public List<OrderItem> Items { get; set; }

    [Required]
    public decimal Total { get; set; }

    [JsonConverter(typeof(JsonStringEnumConverter))]
    [Required]
    public OrderStatus Status { get; set; } = OrderStatus.Pending;

    public string PaymentIntentId { get; set; }
}

File: fotostorio-microservices/ClientApps/Marketing.Ng/src/app/_models/Order.ts
import { Product } from './Product';

export class Order {
    id!: number;
    buyerEmail!: string;
    orderDate!: Date;
    sendToAddress!: Address;
    items!: OrderItem[];
    total!: number;
    status!: string;
}

export class Address {
    firstName: string = '';
    lastName: string = '';
    street!: string;
    secondLine: string = '';
    city!: string;
    county!: string;
    postCode!: string;
}

export class OrderItem {
    id!: number;
    product!: Product;
    quantity!: number;
    total!: number;
}